language-support/hs/bindings/src/DA/Ledger.hs

```diff
@@ -115,3 +119,15 @@ withGetAllTransactionTrees lid party verbose act = do
     let filter = filterEverthingForParty party
     let req = GetTransactionsRequest lid LedgerBegin Nothing filter verbose
     withGetTransactionTrees req act
+
+
+-- Would be nice if the underlying service returned the pid on successful upload.
+uploadDarFileGetPid :: LedgerId -> BS.ByteString -> LedgerService (Either String PackageId)
```

Comment:
That looks like a race condition. What if someone else uploads something in the meantime? Given that we have the package it seems like we can do better even without an API change in the underlying service.

A more important issue: Is this guaranteed to always return a single package id? What if I upload a DAR with multiple packages (e.g. one that has several dependencies). Will they not show up in `listPackages`?

Reply:
You are probably right on both points.
I'll move this function into Test.hs & say it's ok because the caller knows.

Reply:
Thanks! Having this in the tests is totally reasonable.